vpr/src/route/rr_graph_uxsdcxx_serializer.h

```diff
@@ -1543,16 +1542,25 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
     inline void* get_rr_graph_rr_nodes(void*& /*ctx*/) final {
         return nullptr;
     }
-
     void finish_load() final {
+        auto& device_ctx = g_vpr_ctx.mutable_device();
         process_rr_node_indices();
-
         rr_nodes_->init_fan_in();
-
+        /* Create a temp copy to convert from vtr::vector to std::vector
+         * This is required because the ``alloc_and_load_rr_indexed_data()`` function supports only std::vector data
+         * type for ``rr_segments``
+         * Note that this is a dirty fix (to avoid massive code changes)
+         * TODO: The ``alloc_and_load_rr_indexed_data()`` function should embrace ``vtr::vector`` for ``rr_segments``
+         */
+        std::vector<t_segment_inf> temp_rr_segs;
+        device_ctx.rr_segments.reserve(segment_inf_.size());
```

Comment:
@umariqbal-rs Sorry. I think I have written something wrong here. We want to build the ``temp_rr_segs`` vector. However, in the for loop, we are filling the device_ctx.rr_segments.

It should be
```
  std::vector<t_segment_inf> temp_rr_segs;
  temp_rr_segs.reserve(segment_inf_.size());
  for (auto& rr_seg : device_ctx.rr_segments) {
    temp_rr_segs.push_back(rr_seg);
  }
```